generic_server.hh

```diff
@@ -41,6 +42,12 @@ public:
     using connection_process_loop = noncopyable_function<future<> ()>;
     using execute_under_tenant_type = noncopyable_function<future<> (connection_process_loop)>;
     bool _tenant_switch = false;
+    struct cpu_concurrency_t {
+        lw_shared_ptr<named_semaphore> semaphore;
```

Comment:
Why do we need a pointer to the semaphore? And a managed one at that? Could a connection outlive the server?

Reply:
because of dynamic config change, then new one is created and existing connections use the old one

Reply:
Yes, I understood it later. IMO it's less complicated to adjust the units in a static semaphore. e.g. semaphore::consume() to reduce the count.